### DISCUSSION TITLE
Duplicate Prevention Time Range Option

### DISCUSSION THREAD

**Discussion by maerobica** [AzuraCast/AzuraCast#3557] [General]
I quote:
"This specifies the time range (in minutes) of the song history that the duplicate song prevention algorithm should take into account."

Does this mean that if there are no duplication of songs (same file uploaded two times) on the playlist, it (algorithm choosing the songs in random mode) would not play the same song again until the time rage set is exceeded? Or does it apply for duplicated songs only? 

If yes, is there a possibility to adjust the random function not to repeat the songs plaid?

![Screen Shot 2020-12-13 at 01 40 01](https://user-images.githubusercontent.com/75917201/102000138-633bb700-3ce4-11eb-8713-1d1b9c4e51d8.png)

thanks

**Comment by Vaalyn** [comment]
The duplicate prevention meant here is the one that the AutoDJ applies when a playlist is set to `Avoid Duplicate Artists/Titles` to prevent an artist / title+artist from being repeated in that time frame when the AutoDJ builds the upcoming song queue.

**Reply by maerobica** [reply]
Hi Vaalyn, thanks.

Is there a way to set the time for the same song not to be repeated for X time when random function is used for the playback? 
Let's assume there are no duplications.

**Reply by Vaalyn** [reply]
Like I said, that’s what this field is for. Just make sure to also activate the duplicate prevention on the playlist.
